test/es-module/test-esm-experimental-warnings.mjs

```diff
@@ -0,0 +1,31 @@
+import { mustCall } from '../common/index.mjs';
+import { fileURL } from '../common/fixtures.mjs';
+import { match, strictEqual } from 'assert';
+import { spawn } from 'child_process';
+import { execPath } from 'process';
+
+// Verify experimental warnings are printed
+for (
+  const [experiment, arg] of [
+    ['Custom ESM Loaders', `--experimental-loader=${fileURL('es-module-loaders', 'hooks-custom.mjs')}`],
+    ['Network Imports', '--experimental-network-imports'],
+    ['Specifier Resolution', '--experimental-specifier-resolution=node'],
```

Comment:
```suggestion
    [/Custom ESM Loaders/, `--experimental-loader=${fileURL('es-module-loaders', 'hooks-custom.mjs')}`],
    [/Network Imports/, '--experimental-network-imports'],
    [/Specifier Resolution/, '--experimental-specifier-resolution=node'],
```

Reply:
🤔 this seems a bit of a DX pitfall to me. Is there an advantage to the regex here instead of the `new RegExp()` below? (I think the slash syntax is a little more performant, but I think negligibly so)

Reply:
If for whatever reason a special char is added in this string, it would no longer test what we think it's testing. You can either keep strings and use `assert.ok(stderr.includes(…), new Error(…))`, or use regexes and use `assert.match(…, stderr)` – using `new RegExp` is a footgun imo, let's not use it.

Reply:
ah, okay. The alternative with the manual error message seems worse that the maybe pitfall, so I switched the strings to slash-based regexes :)

Reply:
Soo this actually breaks the heck outta the test—for some reason, it never gets passed the internal instance of the ESMLoader, which I'm (now) using as a gate to avoid a redundant print for specifier resolution.

```js
ESMLoader {
  constructor(isInternal = false) {
    if (isInternal) return;

    if (getOptionValue('--experimental-loader')) {
      emitExperimentalWarning('Custom ESM Loaders');
    }
    if (getOptionValue('--experimental-network-imports')) {
      emitExperimentalWarning('Network Imports');
    }
    if (getOptionValue('--experimental-specifier-resolution') === 'node') {
      process.emitWarning(
        'The Node.js specifier resolution flag is experimental. It could change or be removed at any time.',
        'ExperimentalWarning'
      );
    }
  }
}
```